### ISSUE TITLE
[tableio] destructors are noexcept in c++11 (Trac #1661)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1653]
Don't call log_fatal in a destructor.

```text
In file included from
    /home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/IcetrayFwd.
    h:38:0,
from
    /home/dschultz/Documents/combo/trunk/src/tableio/public/tableio/I3TableServ
    ice.h:17,
from /home/dschultz/Documents/combo/trunk/src/tableio/private/tableio/I3TableService.cxx:12:
    /home/dschultz/Documents/combo/trunk/src/tableio/private/tableio/I3TableService
    .cxx: In destructor ‘virtual I3TableService::~I3TableService()’:
/home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/I3Logging.h:181:57:
    warning: throw will always call terminate() [-Wterminate]
##__VA_ARGS__) + " (in " + __PRETTY_FUNCTION__ + ")")
^
/home/dschultz/Documents/combo/trunk/src/tableio/private/tableio/I3TableService.cxx:217:13:
    note: in expansion of macro ‘log_fatal’
log_fatal(
        "Table '%s' is still connected, which means that the "
^~~~~~~~~
/home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/I3Logging.h:181:57:
    note: in C++11 destructors default to noexcept
##__VA_ARGS__) + " (in " + __PRETTY_FUNCTION__ + ")")
^
/home/dschultz/Documents/combo/trunk/src/tableio/private/tableio/I3TableService.cxx:217:13:
    note: in expansion of macro ‘log_fatal’
log_fatal(
        "Table '%s' is still connected, which means that the "
^~~~~~~~~
```

Migrated from https://code.icecube.wisc.edu/ticket/1661
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:12:47", 
    "description": "Don't call log_fatal in a destructor.\n\n{{{\nIn file included from\n    /home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/IcetrayFwd.\n    h:38:0,\nfrom\n    /home/dschultz/Documents/combo/trunk/src/tableio/public/tableio/I3TableServ\n    ice.h:17,\nfrom /home/dschultz/Documents/combo/trunk/src/tableio/private/tableio/I3TableService.cxx:12:\n    /home/dschultz/Documents/combo/trunk/src/tableio/private/tableio/I3TableService\n    .cxx: In destructor \u2018virtual I3TableService::~I3TableService()\u2019:\n/home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/I3Logging.h:181:57:\n    warning: throw will always call terminate() [-Wterminate]\n##__VA_ARGS__) + \" (in \" + __PRETTY_FUNCTION__ + \")\")\n^\n/home/dschultz/Documents/combo/trunk/src/tableio/private/tableio/I3TableService.cxx:217:13:\n    note: in expansion of macro \u2018log_fatal\u2019\nlog_fatal(\n        \"Table '%s' is still connected, which means that the \"\n^~~~~~~~~\n/home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/I3Logging.h:181:57:\n    note: in C++11 destructors default to noexcept\n##__VA_ARGS__) + \" (in \" + __PRETTY_FUNCTION__ + \")\")\n^\n/home/dschultz/Documents/combo/trunk/src/tableio/private/tableio/I3TableService.cxx:217:13:\n    note: in expansion of macro \u2018log_fatal\u2019\nlog_fatal(\n        \"Table '%s' is still connected, which means that the \"\n^~~~~~~~~\n}}}", 
    "reporter": "david.schultz", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067167842669", 
    "component": "combo core", 
    "summary": "[tableio] destructors are noexcept in c++11", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-04-26T20:12:46", 
    "milestone": "", 
    "owner": "jvansanten", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160728T22:48:03: 
* olivas commented:

> In changeset 148875:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="148875"
> changed log_fatal to log_error.  don't throw in a dtor.  closes #1653.
> ```


* olivas changed status from "new" to "closed"
* olivas changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:12:47: 
* olivas changed milestone from "June 2016" to ""
* olivas commented:

> Milestone June 2016 deleted
